src/pages/spots.tsx

```diff
@@ -38,6 +40,12 @@ const SpotsPage = () => {
     enabled: !!spot?.name
   })
 
+  const {data: currentWeather} = useQuery(['current_weather'], () => getCurrentWeather({lat: spot!.latitude, lng: spot!.longitude}), {
+    enabled: !!spot?.name
+  })
```

Comment:
@wondersloth This is the state management for the query using react-query. This is a hook. We're passing it a 'key' and query fn. The query function here I wrote is just a simple wrapper for an axios call. Its a nice pattern I've been using for a long time now. We are using v4: https://tanstack.com/query/v4